### DISCUSSION TITLE
Problems on dev

### DISCUSSION THREAD

**Discussion by thdg** [coqui-ai/TTS#2007] [General Q&A]
Hi,
I am implementing a new language recipe on the dev branch and am running into two problems.
Don't think they warrant a bug report yet since I can not pinpoint the location of the problem.

1)
The recipe does not create a phoneme_cache for the evaluation set, but when the evaluation step comes it tries to load the evaluation items from the cache location. 
This is my dataset config:
dataset_config = BaseDatasetConfig(
    formatter="talromur", meta_file_train="index_train_1000.tsv", meta_file_val="index_eval.tsv", path=os.path.join(output_path, "../salka/")
)
I get the same error if I skip the meta_file_val and let the load_tts_samples function take care of the splitting.

2) 
I am using a special character set for the IPAPhonemes class. It loads well in the beginning but in the evaluation step or somwhere in between it is perhaps reloading the default Tokenizer since I get missing characters error.
The error does not appear when I replace the default phoneme set to my own in TTS/tts/utils/text/characters.py.
This is how I configure the Tokenizer in the recipe:

phonemes = "_:09aAcCDeEfGhiIjJklmMnNoOpPrsStTuvxY"
characters = IPAPhonemes(characters=phonemes)
tokenizer, config = TTSTokenizer.init_from_config(config, characters=characters)

Does anyone have ideas why this might be happening?

**Comment by freezerain** [comment]
1) Do you have correct path in ModelTTSConfig - `phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),`
I remember getting errors related to cache if path is not provided

**Reply by thdg** [reply]
Yes, and I see a lot of files in the cache folder, but only from the train set.

config = GlowTTSConfig(
    batch_size=16,
    eval_batch_size=8,
    num_loader_workers=2,
    num_eval_loader_workers=2,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=1000,
    text_cleaner="basic_cleaners",
    use_phonemes=True,
    phoneme_language="is",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    print_step=25,
    print_eval=False,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
    test_sentences=[
        "Halló, hvernig hefur þú það?",
        "halló hvernig hefur þú það",
    ],
)
